core/commands/filestore.go

```diff
@@ -115,68 +163,70 @@ ERROR:    internal error, most likely due to a corrupt database
 For ERROR entries the error will also be printed to stderr.
 `,
 	},
-	Arguments: []cmds.Argument{
-		cmds.StringArg("obj", false, true, "Cid of objects to verify."),
+	Arguments: []cmdkit.Argument{
+		cmdkit.StringArg("obj", false, true, "Cid of objects to verify."),
 	},
-	Options: []cmds.Option{
-		cmds.BoolOption("file-order", "verify the objects based on the order of the backing file"),
+	Options: []cmdkit.Option{
+		cmdkit.BoolOption("file-order", "verify the objects based on the order of the backing file"),
 	},
-	Run: func(req cmds.Request, res cmds.Response) {
-		_, fs, err := getFilestore(req)
+	Run: func(req oldCmds.Request, res oldCmds.Response) {
+		_, fs, err := getFilestore(req.InvocContext())
 		if err != nil {
-			res.SetError(err, cmds.ErrNormal)
+			res.SetError(err, cmdkit.ErrNormal)
 			return
 		}
 		args := req.Arguments()
 		if len(args) > 0 {
-			out := perKeyActionToChan(args, func(c *cid.Cid) *filestore.ListRes {
+			out := perKeyActionToChan(req.Context(), args, func(c *cid.Cid) *filestore.ListRes {
 				return filestore.Verify(fs, c)
-			}, req.Context())
+			})
 			res.SetOutput(out)
 		} else {
 			fileOrder, _, _ := req.Option("file-order").Bool()
 			next, err := filestore.VerifyAll(fs, fileOrder)
 			if err != nil {
-				res.SetError(err, cmds.ErrNormal)
+				res.SetError(err, cmdkit.ErrNormal)
 				return
 			}
-			out := listResToChan(next, req.Context())
+			out := listResToChan(req.Context(), next)
 			res.SetOutput(out)
 		}
 	},
-	Marshalers: cmds.MarshalerMap{
-		cmds.Text: func(res cmds.Response) (io.Reader, error) {
-			outChan, ok := res.Output().(<-chan interface{})
+	Marshalers: oldCmds.MarshalerMap{
+		oldCmds.Text: func(res oldCmds.Response) (io.Reader, error) {
```

Comment:
so this marshaler now gets called multiple times? (I just noticed there used to be a loop, and now there is not)

Reply:
Marshalers are now called once per value. In the new API we use Encoders, which directly write to a given io.Writer instead of writing to a buffer.

That's why I had to port `ipfs stats bw` to the new API and make use of PostRun to get --poll to work properly (i.e. make it not write the table headers each time)